### PR TITLE
Bugfix/#162 enhanced invitation security

### LINKED ISSUE #162: ROLE_ADMIN cannot view Audits
image

### PR DESCRIPTION
**Pull request by bbortt**
PR contains:
* invitation uniqueness `jhi_user_id` per `project_id`
* invitation frontend uniqueness check by `email`
* cleanup job which removes `Invitation` elder than 14 days
* email update which states the validity of the token
* invitation API security
* minor other test improvements, spelling mistakes fixed

closes #162.

### CODE REVIEW

File: src/main/java/io/github/bbortt/event/planner/repository/InvitationRepository.java [hunk 1]
```diff
@@ -23,4 +25,8 @@
     void assignUserToInvitation(@Param("userId") Long userId, @Param("token") String token);
 
     Optional<Invitation> findByToken(String token);
+
+    Optional<Invitation> findOnyByEmailAndProjectId(String email, Long projectId);
+
+    List<Invitation> findAllByAcceptedIsFalseAndAndTokenIsNotNullAndCreatedDateBefore(Instant dateTime);
```
Comment (zeroplexer) [inline src/main/java/io/github/bbortt/event/planner/repository/InvitationRepository.java:31]: 2x `And`

File: src/main/java/io/github/bbortt/event/planner/web/rest/InvitationResource.java [hunk 2]
```diff
@@ -113,6 +129,7 @@ public InvitationResource(InvitationService invitationService, MailService mailS
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and the list of invitations in body.
      */
     @GetMapping("/invitations")
+    @PreAuthorize("hasAuthority(\"" + AuthoritiesConstants.ADMIN + "\")")
```
Comment (zeroplexer) [inline src/main/java/io/github/bbortt/event/planner/web/rest/InvitationResource.java:132]: Secretary sollte auch Zugriff haben

**Review by zeroplexer — CHANGES_REQUESTED** [review]
Small changes

File: src/main/java/io/github/bbortt/event/planner/web/rest/InvitationResource.java (hunk 2 above)
Comment (bbortt) [reply src/main/java/io/github/bbortt/event/planner/web/rest/InvitationResource.java:132]: nein, dies ist der applikatorische `ADMIN` und nur dieser darf _alle_ `Invitation` auflisten.

**Pull request merged**